code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -52,29 +52,41 @@
 	icon_state = "down"
 	anchored = FALSE
 	resistance_flags = NONE
-	var/foldabletype = /obj/item/roller
+	/// Internal storage for the roller item that created us
+	var/obj/item/roller
 
-/obj/structure/bed/roller/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/roller/robo))
-		var/obj/item/roller/robo/R = W
-		if(R.loaded)
-			to_chat(user, span_warning("You already have a roller bed docked!"))
-			return
+/obj/structure/bed/roller/Initialize(mapload, obj/item/roller/roller)
+	. = ..()
+	if(!roller)
+		roller = new /obj/item/roller(src)
+	src.roller = roller
+	roller.forceMove(src)
```

Comment:
roller already exists inside of src at this point

Reply:
Not if it didn't make a new one